Request/ParamFetcher.php

```diff
@@ -48,14 +48,14 @@ class ParamFetcher implements ParamFetcherInterface
      * Initializes fetcher.
      *
      * @param ParamReader                 $paramReader
-     * @param Request                     $request
+     * @param RequestStack                $requestStack
      * @param ValidatorInterface|LegacyValidatorInterface $validator
      * @param ViolationFormatterInterface $violationFormatter
      */
-    public function __construct(ParamReader $paramReader, Request $request, ViolationFormatterInterface $violationFormatter, $validator = null)
+    public function __construct(ParamReader $paramReader, RequestStack $requestStack, ViolationFormatterInterface $violationFormatter, $validator = null)
     {
         $this->paramReader        = $paramReader;
-        $this->request            = $request;
+        $this->request            = $requestStack->getCurrentRequest();
```

Comment:
please don't retrieve the request in the constructor. Store the RequestStack and get the request on demand.
This will allow you to move the service to the container scope (the request scope is deprecated in 2.8)